CMU-313/spring23-nodebb-team-hecl-io
# Add Resolve Button Feature (without banner indicator)

**Description:** This branch implements the resolve button feature. The user of NodeBB is now able to toggle between resolving and unresolving a post using this button. 

**What is the feature?**
A button on NodeBB posts that allows a user to mark a post as resolved. Once the post has been resolved, it can also be toggled to unresolved using the same button. 

**How you implemented the solution?**
To learn how NodeBB works, I followed the bookmark feature that already existed in NodeBB. I then followed that implementation to develop the necessary frontend and backend changes to create the resolve toggle button. 

**Does it impact any other area of the project?**
The branch touches 11 files in the project. 

**Any changes in the UI (Screenshots)?**
The resolve and unresolve buttons on posts are UI changes as seen in the screenshots below. 
<img width="639" alt="Screenshot 2023-02-18 at 6 29 49 PM" src="https://user-images.githubusercontent.com/66187015/219904011-43ef5353-ac6a-4440-af41-e6a8eda01635.png">
<img width="637" alt="Screenshot 2023-02-18 at 6 30 01 PM" src="https://user-images.githubusercontent.com/66187015/219904019-4ee86c7c-6bd0-4d70-84d6-f35462a3c1c5.png">



**How to test the feature (A test scenario or any new setup)?**

There are no test scenarios setup for this yet. However, it passes `npm run lint`. Also, we see that the feature works through using the actual NodeBB website.
